tests/RankedPairsGraphTest.php

```diff
@@ -0,0 +1,156 @@
+use PivotLibre\Tideman\RankedPairsCalculator;
+use Fhaculty\Graph\Graph;
+use Fhaculty\Graph\Vertex;
+use Fhaculty\Graph\Set\Vertices;
+
+class RankedPairsGraphTest extends TestCase
+{
+    private const ALICE_ID = "A";
+    private const ALICE_NAME = "Alice";
+    private const BOB_ID = "B";
+    private const BOB_NAME = "Bob";
+    private const CLAIRE_ID = "C";
+    private const CLAIRE_NAME = "Claire";
+    private $alice;
+    private $bob;
+    private $claire;
+    private $instance;
+
+    protected function setUp() :void
+    {
+        $this->alice = new Candidate(self::ALICE_ID, self::ALICE_NAME);
+        $this->bob = new Candidate(self::BOB_ID, self::BOB_NAME);
+        $this->claire = new Candidate(self::CLAIRE_ID, self::CLAIRE_NAME);
+        $this->instance = new RankedPairsGraph();
+    }
+
+    public function testAddCandidateToGraph() : void
+    {
+        //assert empty Graph to start with
+        $this->assertEquals(0, $this->instance->getGraph()->getVertices()->count());
+        $vertex = $this->instance->addCandidateToGraph($this->alice);
+        //assert added to underlying graph
+        $this->assertEquals(1, $this->instance->getGraph()->getVertices()->count());
+        //assert candidate was attached to Vertex
+        $this->assertEquals($this->alice, $vertex->getAttribute(RankedPairsGraph::CANDIDATE_ATTRIBUTE_NAME));
+
+        //try adding it again. The underlying Graph should remain unchanged
+        $secondVertex = $this->instance->addCandidateToGraph($this->alice);
+        //assert underlying graph is still the same
+        $this->assertEquals(1, $this->instance->getGraph()->getVertices()->count());
+        //assert candidate is still attached to Vertex
+        $this->assertEquals($this->alice, $vertex->getAttribute(RankedPairsGraph::CANDIDATE_ATTRIBUTE_NAME));
+        //the second addition of a candidate should return the exact same object as the one returned by the
+        //first addition
+        $this->assertSame($vertex, $secondVertex);
+    }
+
+    public function testGetCandidatesFromOneLengthVertices() : void
+    {
+        //add just one vertex
+        $this->instance->addCandidateToGraph($this->alice);
+        $vertices = $this->instance->getGraph()->getVertices();
+        $expected = new CandidateList($this->alice);
+        $actual = $this->instance->getCandidatesFromVertices($vertices);
+        $this->assertEquals($expected, $actual);
+    }
+
+    public function testGetCandidatesFromTwoLengthVertices() : void
+    {
+        //add two vertices
+        $this->instance->addCandidateToGraph($this->alice);
+        $this->instance->addCandidateToGraph($this->bob);
+        $vertices = $this->instance->getGraph()->getVertices();
+        $expected = new CandidateList($this->alice, $this->bob);
+        $actual = $this->instance->getCandidatesFromVertices($vertices);
+        $this->assertEquals($expected, $actual);
+    }
+
+    public function testVertexIsInACycle() : void
+    {
```

Comment:
This is a very trivial cycle. It would be good to expand this so that we verify we are correctly detecting non-trivial cycles.